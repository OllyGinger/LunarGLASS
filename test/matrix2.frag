#version 130

//#define TEST_POST_110

uniform mat3 colorTransform;
varying vec3 Color;
uniform mat4 m, n;

#ifdef TEST_POST_110
uniform mat4x3 um43;
uniform mat3x4 un34;
#else
uniform mat4 um43;
uniform mat4 un34;
#endif

varying vec4 v;

#ifdef TEST_POST_110
varying vec3 u;
#else
varying vec4 u;
#endif

void main()
{
    
#ifdef TEST_POST_110
    mat3x4 m34 = outerProduct(v, u);
#else
    mat4 m34 = mat4(v.x*u.x, v.x*u.y, v.x*u.z, v.x*u.w, 
                    v.y*u.x, v.y*u.y, v.y*u.z, v.y*u.w, 
                    v.z*u.x, v.z*u.y, v.z*u.z, v.z*u.w, 
                    v.w*u.x, v.w*u.y, v.w*u.z, v.w*u.w);
#endif

    m34 += mat4(4.3);

    gl_FragColor = vec4(Color, 1.0);
    gl_FragColor *= um43;

    m34 *= v.x;

#ifdef TEST_POST_110
    mat4 m44 = mat4(un34);
#else
    mat4 m44 = mat4(4.7);
#endif

    m44 *= um43;

    //m44 = matrixCompMult(m44, m44);

    gl_FragColor += (-m44) * u;

    gl_FragColor *= matrixCompMult(m44, m44);
}
