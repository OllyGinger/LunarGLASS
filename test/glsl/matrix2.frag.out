
***Unsupported functionality: Matrix constants
compiling matrix2.frag...
uniform mat3 colorTransform ; 
varying vec3 Color ; 
uniform mat4 m , n ; 
uniform mat4 um43 ; 
uniform mat4 un34 ; 
varying vec4 v ; 
varying vec4 u ; 
void  main (){
mat4 m34 = mat4 ( v . x * u . x , v . x * u . y , v . x * u . z , v . x * u . w , v . y * u . x , v . y * u . y , v . y * u . z , v . y * u . w , v . z * u . x , v . z * u . y , v . z * u . z , v . z * u . w , v . w * u . x , v . w * u . y , v . w * u . z , v . w * u . w ) ; m34 += mat4 ( 4.300000 ) ; gl_FragColor = vec4 ( Color , 1.000000 ) ; gl_FragColor *= um43 ; m34 *= v . x ; mat4 m44 = mat4 ( 4.700000 ) ; m44 *= um43 ; gl_FragColor += - m44 * u ; gl_FragColor *= matrixCompMult ( m44 , m44 ) ; }



(
(declare (in ) vec4 u)
(declare (uniform ) mat4 um43)
(declare (in ) vec3 Color)
(declare (out ) vec4 gl_FragColor)
(function main
  (signature void
    (parameters
    )
    (
      (declare (temporary ) vec4 vec_ctor)
      (assign (constant bool (1)) (w) (var_ref vec_ctor)  (constant float (1.000000)) ) 
      (assign (constant bool (1)) (xyz) (var_ref vec_ctor)  (var_ref Color) ) 
      (declare (temporary ) mat4 assignment_tmp)
      (assign (constant bool (1)) () (var_ref assignment_tmp)  (expression mat4 * (constant mat4 (4.700000 0.000000 0.000000 0.000000 0.000000 4.700000 0.000000 0.000000 0.000000 0.000000 4.700000 0.000000 0.000000 0.000000 0.000000 4.700000)) (var_ref um43) ) ) 
      (assign (constant bool (1)) (xyzw) (var_ref gl_FragColor)  (expression vec4 + (expression vec4 * (var_ref vec_ctor) (var_ref um43) ) (expression vec4 * (expression mat4 neg (var_ref assignment_tmp) ) (var_ref u) ) ) ) 
      (declare () mat4 z)
      (assign (constant bool (1)) (xyzw) (array_ref (var_ref z) (constant int (0)) )  (expression vec4 * (array_ref (var_ref assignment_tmp) (constant int (0)) ) (array_ref (var_ref assignment_tmp) (constant int (0)) ) ) ) 
      (assign (constant bool (1)) (xyzw) (array_ref (var_ref z) (constant int (1)) )  (expression vec4 * (array_ref (var_ref assignment_tmp) (constant int (1)) ) (array_ref (var_ref assignment_tmp) (constant int (1)) ) ) ) 
      (assign (constant bool (1)) (xyzw) (array_ref (var_ref z) (constant int (2)) )  (expression vec4 * (array_ref (var_ref assignment_tmp) (constant int (2)) ) (array_ref (var_ref assignment_tmp) (constant int (2)) ) ) ) 
      (assign (constant bool (1)) (xyzw) (array_ref (var_ref z) (constant int (3)) )  (expression vec4 * (array_ref (var_ref assignment_tmp) (constant int (3)) ) (array_ref (var_ref assignment_tmp) (constant int (3)) ) ) ) 
      (assign (constant bool (1)) (xyzw) (var_ref gl_FragColor)  (expression vec4 * (var_ref gl_FragColor) (var_ref z) ) ) 
    ))

)


)